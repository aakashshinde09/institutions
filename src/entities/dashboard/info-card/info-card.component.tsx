import { Skeleton } from '@mantine/core'
import { cn } from '@shared/lib/utils'
import { CaptionItem } from '@shared/ui/caption-item'
import type { ReactElement } from 'react'

interface IInfoCardProperties {
  data: {
    title: string
    value: string
  }[]
  className?: string
  chart?: ReactElement
  showSkeleton?: boolean
}

export const InfoCard = ({
  data,
  className,
  chart,
  showSkeleton,
}: IInfoCardProperties) => {
  return (
    <article
      className={cn(
        'flex rounded-[1.25rem] bg-white px-4 py-8 justify-between gap-4 max-md:p-4',
        className,
      )}
    >
      <div className="flex w-full flex-col justify-end space-y-6">
        {showSkeleton
          ? Array.from({ length: data.length }).map((_, index) => (
              <Skeleton key={index} width={chart ? '100%' : '60%'} height="3.5rem" />
            ))
          : data.map((item, index) => (
              <CaptionItem key={index} title={item.title} value={item.value} />
            ))}
      </div>
      {chart}
    </article>
  )
}
