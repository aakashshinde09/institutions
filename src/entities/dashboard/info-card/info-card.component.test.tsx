import { expect, test, describe } from 'vitest'
import { render } from '@testing-library/react'
import { InfoCard } from './info-card.component'

const mock = [
  {
    title: 'Amount staked',
    value: `10 stEth`,
  },
  {
    title: 'Percentage of staking rewards donated',
    value: `17 %`,
  },
]

describe('<InfoCard/>', () => {
  test('Renders correctly', () => {
    const component = render(<InfoCard data={mock} showSkeleton={false} />)

    expect(component).toMatchSnapshot()
  })
})
