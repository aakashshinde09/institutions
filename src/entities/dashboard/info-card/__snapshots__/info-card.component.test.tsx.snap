// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<InfoCard/> > Renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="flex rounded-[1.25rem] bg-white px-4 py-8 justify-between gap-4 max-md:p-4"
      >
        <div
          class="flex w-full flex-col justify-end space-y-6"
        >
          <div
            class="truncate"
          >
            <p
              class="font-semibold max-md:text-sm max-md:font-normal"
            >
              Amount staked
            </p>
            <span
              class="mt-2 text-2xl font-bold max-md:text-base"
            >
              10 stEth
            </span>
          </div>
          <div
            class="truncate"
          >
            <p
              class="font-semibold max-md:text-sm max-md:font-normal"
            >
              Percentage of staking rewards donated
            </p>
            <span
              class="mt-2 text-2xl font-bold max-md:text-base"
            >
              17 %
            </span>
          </div>
        </div>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="flex rounded-[1.25rem] bg-white px-4 py-8 justify-between gap-4 max-md:p-4"
    >
      <div
        class="flex w-full flex-col justify-end space-y-6"
      >
        <div
          class="truncate"
        >
          <p
            class="font-semibold max-md:text-sm max-md:font-normal"
          >
            Amount staked
          </p>
          <span
            class="mt-2 text-2xl font-bold max-md:text-base"
          >
            10 stEth
          </span>
        </div>
        <div
          class="truncate"
        >
          <p
            class="font-semibold max-md:text-sm max-md:font-normal"
          >
            Percentage of staking rewards donated
          </p>
          <span
            class="mt-2 text-2xl font-bold max-md:text-base"
          >
            17 %
          </span>
        </div>
      </div>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
